src/EduManagementLab.Web/Pages/Users/Create.cshtml.cs

```diff
@@ -43,14 +43,20 @@ public IActionResult OnGet()
         // To protect from overposting attacks, see https://aka.ms/RazorPagesCRUD
         public async Task<IActionResult> OnPostAsync()
         {
-            if (!ModelState.IsValid)
+            if (ModelState.IsValid)
             {
-                return Page();
+                try
+                {
+                    _userService.CreateUser(Displayname, FirstName, LastName, Email);
+                    return RedirectToPage("./Index");
+                }
+                catch (Exception)
+                {
+                    ModelState.AddModelError("Email", "User already exist");
```

Comment:
You dont know that the Exception is thrown because of User already exists.
It can be another error.